ast/src/test/java/com/graphicsfuzz/common/util/ParseHelperTest.java

```diff
@@ -566,17 +566,61 @@ private String getStringFromInputStream(InputStream strippedIs) throws IOExcepti
   }
 
   @Test
```

Comment:
Thanks for these tests.  Can you add something to PrettyPrinterVisitorTest to check that an example that uses non-literal array sizes pretty-prints to itself?

Also, do we have any tests in this PR that check the folding does the right thing?  If not it would be good to have them.